webapp/src/immersivereader.tsx

```diff
@@ -46,29 +62,98 @@ function beautifyText(content: string): string {
         );
     }
 
-    // Remove ** around bold text, replace with "
-    function cleanBold(content: string): string {
+    // If the ```block ``` annotation is used to display a block in
+    // a modal, remove it
+    function cleanHintText(content: string): string {
         return content.replace(
-            /\*\*([^*]+)\*\*/gu,
-            (matched, word, offset, s) => lf("\"{0}\"", word)
+            /^```block$[^`]*^```$/gum,
+            ""
+        );
+    }
+
+    // Remove ** and __ around bold text, replace with " if it's only one word
+    function cleanBold(content: string): string {
+
+        let singleWordBold = content.replace(
+            /\*\*([^*^\s]+)\*\*|__([^\s^_]+)__/gu,
+            (matched, boldStar, boldUnder, offset, s) => {
+                return "\"" + lf("{0}", (boldStar ? boldStar : boldUnder)) + "\""
+            }
+        )
+
+        return singleWordBold.replace(
+            /\*\*([^*]+)\*\*|__([^_]+)__/gu,
+            (matched, boldStar, boldUnder, offset, s) => {
+                return boldStar ? lf("{0}", boldStar) : lf ("{0}", boldUnder);
+            }
         );
     }
 
     // replace horizontal rules with new lines
     function cleanHorizontalRule(content: string): string {
         return content.replace(
-            /^\* \* \*$/gum,
-            "- - -"
+            /^\* \* \*$|^- - -$|^<hr\/>$/gum,
```

Comment:
I might be misreading this one but did this not get included in the build? looks like it still has the behavior I had it as with swapping horizontal rule for `- - -`? 
![image](https://user-images.githubusercontent.com/5615930/112515853-e6dd5900-8d53-11eb-959d-8e4b3d283911.png) from galga step 2

Reply:
OH thanks for finding that i will investigate 🔍